### DISCUSSION TITLE
what's the type of a WalletClient that extend publicActions?

### DISCUSSION THREAD

**Discussion by Stachelabs** [wevm/viem#1463] [Question] [answered]
Hello I'm pretty new to typescript but I decided to make my project with viem over ethers.js in typescript.

I have a walletClient that extend to publicActions:

```typescript

L2Wallet = createWalletClient({
      account: accounts[i],
      chain: scroll,
      transport: http(process.env.SCROLL_MAINNET_RPC_URL),
    }).extend(publicActions);
```

But I need the type.

So far i've created an interface but i'm sure there is already a built in type in viem:

```
interface SuperWallet extends WalletClient, PublicActions {}
```

If anyone can help, than you very much

**Comment by jxom** [comment, marked as answer]
Here you go!

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3nAggYxxAVwDsYAaOAYQBtgUTyKALAQ2CPIAUCAjGnXGMAhEAzuQBKYHAGUcjFCGbkAKlGajIscgHVmVKihgChouAF84AMygQQcAEQA3WiHsBuAFAeUAD03wkVDhpAlQoXX1DaloSAB4POAQ1DWh4Xxg6ABMROFV1EX84AF5c5ILU0gS4OVYiOHSsnKZauAAfOGJMlEs2FEziyhY2No6iLp6iPsrE5jxCEnqfDLGc3HxieHbO7t7+krX5zdHx3cqAPgHouhh4xJgy-2nqofYq2fX6KskZOQUlKq4vGA-BwgmEIli93yj2etXI70OFwAZHAIgYjKCTBCamx4XMNmcPGdPB58KJ4DgaNcBugLMwciEwmiolSSF5KTEYAA6ERZPKiWZgogeAD0IsSEoAegB+dms7kAc0M-JEgpMovFErgMqAA

**Reply by Stachelabs** [reply]
@jxom thank you very much looks like this is more complete
